mrobosub_sim/mrobosub_sim/hal.py

```diff
@@ -416,16 +436,23 @@ def incoming_callback(self, client: net.socket):
     def run(self):
         self.outgoing_server_thread.start()
         self.incoming_server_thread.start()
-        rospy.spin()
+        rclpy.spin(self)
         self.outgoing_server.stop()
         self.incoming_server.stop()
-        for client in self.clients:
+        for client in self.net_clients:
             client.stop()
         self.outgoing_server_thread.join()
         self.incoming_server_thread.join()
-        for client in self.clients:
+        for client in self.net_clients:
             client.thread.join()
 
 
+def run():
+    rclpy.init()
+    global node
```

Comment:
Why does this need to be global?

Reply:
It doesn't I was gonna do that instead of passing node to everything but I decided that was more pythonic

Reply:
Actually there is still one place its needed in SensorData.unpack